### ISSUE TITLE
DoWhen Infinite Loop - Earing gorge

### ISSUE THREAD

**Issue by bartolije** [EchoTiger/NewQuestingPack#1]
The bot is stuck in a loop, doing a short go-back on few meter.
HB log spam the following:
```
[DoWhen-7b5abb(debug)] Stopped executing SellExtraConsumables hook
[DoWhen-7b5abb(debug)] Executing SellExtraConsumables hook
```

The info tab:

```
Goal: InteractWith-7b5abb: In Progress (no associated quest)
Interacting with Benêt  [Ref: "Searing Gorge ($Rev: 6041 $)" @line 205]
```

[5596 2017-05-13 23.23.txt](https://github.com/EchoTiger/NewQuestingPack/files/998957/5596.2017-05-13.23.23.txt)

**Comment by EchoTiger** [comment]
I'm assuming other players around you have the Lunk NPC, and the bot is detecting theirs - which is causing interaction with yours to fail since yours doesn't have any venom to obtain.

The DoWhen code is there to get the spider venom from Lunk.
I've set it to only detect the Lunk NPC that is  "CreatedByUnit == Me" which should hopefully fix this.

I can't test it, so it will need to be checked.
I've pushed a "fix" for it on commit https://github.com/EchoTiger/NewQuestingPack/commit/06c1b8f7b4b42e28104aac70b0739edec2633978
